jujugui/static/gui/src/app/app.js

```diff
@@ -774,6 +774,10 @@ YUI.add('juju-gui', function(Y) {
                 this, this.db, this.env, service)}
             createMachinesPlaceUnits={utils.createMachinesPlaceUnits.bind(
                 this, this.db, this.env, service)}
+            destroyService={utils.destroyService.bind(
+                this, this.db, this.env, service)}
+            clearState={utils.clearState.bind(
```

Comment:
I'm curious as to why you created a clearState method in utils instead of just calling it on `topo` which you're passing in below.